cpp/ql/src/semmle/code/cpp/XML.qll

```diff
@@ -2,27 +2,30 @@
  * Provides classes and predicates for working with XML files and their content.
  */
 
-import semmle.code.cpp.Location
+import semmle.files.FileSystem
 
 /** An XML element that has a location. */
 abstract class XMLLocatable extends @xmllocatable {
   /** Gets the source location for this element. */
   Location getLocation() { xmllocations(this, result) }
 
   /**
-   * Holds if this element has the specified location information,
-   * including file path, start line, start column, end line and end column.
+   * Holds if this element is at the specified location.
+   * The location spans column `startcolumn` of line `startline` to
+   * column `endcolumn` of line `endline` in file `filepath`.
+   * For more information, see
+   * [Locations](https://help.semmle.com/QL/learn-ql/ql/locations.html).
    */
   predicate hasLocationInfo(
     string filepath, int startline, int startcolumn, int endline, int endcolumn
   ) {
     exists(File f, Location l | l = this.getLocation() |
-      locations_default(l, unresolveElement(f), startline, startcolumn, endline, endcolumn) and
+      locations_default(l, f, startline, startcolumn, endline, endcolumn) and
```

Comment:
@igfoo, do you agree this is tolerable for now? If it _must_ use `unresolveElement` in the future, we can come back and add a pyrameter file.

Reply:
Currently it'll unresolve to itself, so I think it's OK for now.